lib/if.c

```diff
@@ -1106,14 +1105,23 @@ DEFPY_NOSH (interface,
 	 * interface is found, then a new one should be created on the default
 	 * VRF.
 	 */
-	VRF_GET_ID(vrf_id, vrfname, false);
-	ifp = if_lookup_by_name_all_vrf(ifname);
-	if (ifp && ifp->vrf_id != vrf_id) {
+	VRF_GET_INSTANCE(vrf, vrfname, false);
```

Comment:
If we are adding a new VRF_GET_INSTANCE why are we not replacing a bunch of the uses of VRF_GET_ID with it?  There are multiple places that it would just slot in.

Reply:
some improvements is done hre.